src/lib.rs

```diff
@@ -552,6 +555,9 @@ impl<T: Counter> Histogram<T> {
             if other_count != T::zero() {
                 let other_value = other.value_for(i);
                 if self.count_at(other_value).unwrap() < other_count {
+                    // TODO Perhaps we should saturating sub here? Or expose some form of
+                    // pluggability so users could choose to error or saturate? Both seem useful.
+                    // It's also sort of inconsistent with overflow, which now saturates.
```

Comment:
Hmm, yes, it's a good question.. 
One thing to do would be to have a variant in `SubtractionError` (and possibly also `AdditionError`) called `PartialResult` (or something like that), which contains the `Histogram` when using saturating subtract/add. That way the user can either chose to deal with it as an error, or they can match that case and ignore the error.